src/tools/target-docs/src/render.rs

```diff
@@ -0,0 +1,257 @@
+use std::{fs, path::Path};
+
+use crate::TargetInfo;
+
+/// Renders a single target markdown file from the information obtained.
+pub fn render_target_md(target: &TargetInfo) -> String {
+    let render_header_option_bool = |bool| match bool {
+        Some(true) => "Yes",
+        Some(false) => "No",
+        None => "?",
+    };
+
+    let mut doc = format!(
+        "# {}\n\n**Tier: {}**\n\n**std: {}**\n\n**host tools: {}**\n\n",
+        target.name,
+        match target.metadata.tier {
+            Some(1) => "1",
+            Some(2) => "2",
+            Some(3) => "3",
+            _ => "UNKNOWN",
+        },
+        render_header_option_bool(target.metadata.std),
+        render_header_option_bool(target.metadata.host_tools),
+    );
+
+    let mut section = |name: &str, content: &str| {
+        doc.push_str("## ");
+        doc.push_str(name.trim());
+        doc.push('\n');
+        doc.push_str(content.trim());
+        doc.push_str("\n\n");
+    };
+
+    let maintainers_content = if target.maintainers.is_empty() {
+        "This target does not have any maintainers!".to_owned()
+    } else {
+        format!(
+            "This target is maintained by:\n{}",
+            target
+                .maintainers
+                .iter()
+                .map(|maintainer| {
+                    let maintainer = if maintainer.starts_with('@') && !maintainer.contains(' ') {
+                        format!(
+                            "[@{0}](https://github.com/{0})",
+                            maintainer.strip_prefix("@").unwrap()
+                        )
+                    } else {
+                        maintainer.to_owned()
+                    };
+
+                    format!("- {maintainer}")
+                })
+                .collect::<Vec<_>>()
+                .join("\n")
+        )
+    };
+
+    section("Maintainers", &maintainers_content);
+
+    for section_name in crate::SECTIONS {
+        let value = target.sections.iter().find(|(name, _)| name == section_name);
+
+        let section_content = match value {
+            Some((_, value)) => value.clone(),
+            None => "Unknown.".to_owned(),
+        };
+        section(section_name, &section_content);
+    }
+
+    let cfg_text = target
+        .target_cfgs
+        .iter()
+        .map(|(key, value)| format!("- `{key}` = `{value}`"))
+        .collect::<Vec<_>>()
+        .join("\n");
+    let cfg_content =
+        format!("This target defines the following target-specific cfg values:\n{cfg_text}\n");
+
+    section("cfg", &cfg_content);
+
+    doc
+}
+
+/// Replaces inner part of the form
+/// `<!-- {section_name} SECTION START --><!-- {section_name} SECTION END -->`
+/// with replacement`.
+fn replace_section(prev_content: &str, section_name: &str, replacement: &str) -> Result<String> {
+    let magic_summary_start = format!("<!-- {section_name} SECTION START -->");
+    let magic_summary_end = format!("<!-- {section_name} SECTION END -->");
```

Comment:
These can probably be `const`s

Reply:
i dont think so

Reply:
Nvm, for some reason I thought they were just constant slices 🤦